java/src/test/java/com/codurance/training/tasks/ApplicationTest.java

```diff
@@ -15,113 +16,114 @@
-                "secrets",
-                "    [x] 1: Eat more donuts.",
-                "    [ ] 2: Destroy all humans.",
-                "",
-                "training",
-                "    [x] 3: Four Elements of Simple Design",
-                "    [ ] 4: SOLID",
-                "    [x] 5: Coupling and Cohesion",
-                "    [x] 6: Primitive Obsession",
-                "    [ ] 7: Outside-In TDD",
-                "    [ ] 8: Interaction-Driven Design",
-                ""
-        );
-
-        execute("quit");
-    }
-
-    private void execute(String command) throws IOException {
-        read(PROMPT);
-        write(command);
-    }
-
-    private void read(String expectedOutput) throws IOException {
-        int length = expectedOutput.length();
-        char[] buffer = new char[length];
-        outReader.read(buffer, 0, length);
-        assertThat(String.valueOf(buffer), is(expectedOutput));
-    }
-
-    private void readLines(String... expectedOutput) throws IOException {
-        for (String line : expectedOutput) {
-            read(line + lineSeparator());
-        }
-    }
-
-    private void write(String input) {
-        inWriter.println(input);
-    }
-
-    private boolean stillRunning() {
-        return applicationThread != null && applicationThread.isAlive();
-    }
+		execute("add project training");
+		execute("add task training asd Four Elements of Simple Design");
+		execute("add task training sdf SOLID");
+		execute("add task training dfg Coupling and Cohesion");
+		execute("add task training fgh Primitive Obsession");
+		execute("add task training hjk Outside-In TDD");
+		execute("add task training jkl Interaction-Driven Design");
+
+		execute("check abc");
+		execute("check asd");
+		execute("check dfg");
+		execute("check fgh");
+		execute("uncheck fgh");
+		execute("deadline abc 02-03-2023");
+		execute("delete jkl");
+		execute("view by project");
+		readLines(
+				"secrets",
+				"    [x] abc: Eat more donuts. ",
+				"    [ ] def: Destroy all humans. ",
+				"",
+				"training",
+				"    [x] asd: Four Elements of Simple Design ",
+				"    [ ] sdf: SOLID ",
+				"    [x] dfg: Coupling and Cohesion ",
+				"    [ ] fgh: Primitive Obsession ",
+				"    [ ] hjk: Outside-In TDD ",
+				""
+		);
+
+		execute("quit");
+	}
+
+	private void execute(String command) throws IOException {
+		read(PROMPT);
+		write(command);
+	}
+
+	private void read(String expectedOutput) throws IOException {
+		int length = expectedOutput.length();
+		char[] buffer = new char[length];
+		outReader.read(buffer, 0, length);
+		assertThat(String.valueOf(buffer), is(expectedOutput));
+	}
+
+	private void readLines(String... expectedOutput) throws IOException {
+		for (String line : expectedOutput) {
+			read(line + lineSeparator());
+		}
+	}
+
+	private void write(String input) {
+		inWriter.println(input);
+	}
+
+	private boolean stillRunning() {
+		return applicationThread != null && applicationThread.isAlive();
+	}
```

Comment:
Keep the existing test and add new tests on top of it
Don't update or remove the old ones

Reply:
I had to change change abhilash, as we are passing idString when we add task